azurelinuxagent/ga/cgroupconfigurator.py

```diff
@@ -508,20 +517,25 @@ def _reset_cpu_quota(unit_name):
             over this setting.
             """
             log_cgroup_info("Resetting {0}'s CPUQuota".format(unit_name), send_event=False)
-            CGroupConfigurator._Impl._try_set_cpu_quota(unit_name, "infinity") # systemd expresses no-quota as infinity, following the same convention
-            log_cgroup_info('Current CPUQuota: {0}'.format(systemd.get_unit_property(unit_name, "CPUQuotaPerSecUSec")))
+            if CGroupConfigurator._Impl._try_set_cpu_quota(unit_name, "infinity"): # systemd expresses no-quota as infinity, following the same convention
+                try:
+                    log_cgroup_info('Current CPUQuota: {0}'.format(systemd.get_unit_property(unit_name, "CPUQuotaPerSecUSec")))
+                except Exception as e:
+                    log_cgroup_warning('Failed to get current CPUQuotaPerSecUSec after reset: {0}'.format(ustr(e)))
 
         # W0238: Unused private member `_Impl.__try_set_cpu_quota(quota)` (unused-private-member)
         @staticmethod
         def _try_set_cpu_quota(unit_name, quota):  # pylint: disable=unused-private-member
             try:
                 current_cpu_quota = CGroupConfigurator._Impl._get_current_cpu_quota(unit_name)
                 if current_cpu_quota == quota:
-                    return
+                    return True
                 quota = quota if quota != "infinity" else ""  # no-quota expressed as empty string while setting property
```

Comment:
it's possible for _get_current_cpu_quota(unit) to return "unknown" which isn't explicitly checked here. Is it expected that quota is reset if _get_current_cpu_quota returns 'unknown'?

This is unrelated to these changes, but noticed it as I was stepping through the unit test

Reply:
As this is unrelated to these changes, we can discuss offline.